components/brave_ads/core/internal/account/tokens/payment_tokens/payment_token_value_util.cc

```diff
@@ -111,6 +111,11 @@ PaymentTokenList PaymentTokensFromValue(const base::Value::List& list) {
 
     // Ad type
     if (const auto* const ad_type = dict->FindString(kAdTypeKey)) {
+      if (*ad_type == "inline_content_ad" ||
+          *ad_type == "promoted_content_ad") {
+        BLOG(0, "Dropping legacy ad type payment token");
```

Comment:
The goal of `NOTREACHED/CHECK(false)` is to prevent security bugs and crash the browser instead.

Is it true that an invalid/unsupported ad type is a security issue so we should crash the browser in this case? I suggest you to reconsider the current impl of `ToMojomAdType` and make it return optional instead.